src/stores/students.ts

```diff
@@ -2,41 +2,73 @@ import { defineStore } from 'pinia'
 
 import { api } from '@/services'
 import type { Student } from '@/services/students'
+import { getStudentsWithMetadata } from '@/helpers/metadataMappers'
+import { removeStudentsMetadata } from '@/helpers/metadataRemovers'
+
+export type StudentWithMetadata = {
+  [T in keyof Student]: {
+    value: Student[T]
+    isValidated: boolean
+  }
+}
```

Comment:
why not:
```suggestion
export type StudentWithMetadata =  {
  value: Student
  isValidated: boolean
}
```
it seems to fix the 2 ts errors that you mentioned